### ISSUE TITLE
TSP143III (Ethernet) UnsupportedModelError on status check and print

### ISSUE THREAD

**Issue by matthewparavati** [star-micronics/react-native-star-io10#132]
## Description
We have 2 client reports of this issue and I have limited data and just hoping someone can point me in the right direction of what _might_ be going wrong in our implementation for TSP143III (Ethernet) printers, as Wifi, USB, and Bluetooth seems to work fine.

I do not have a TSP143III (Ethernet) printer to test on, at the moment, unfortunately, and the 650II ethernet printer I do have is working. Along with TSP143 Bluetooth, USB, and Wifi printers are working.

For our clients, though, it seems like for their ethernet printers, the printer status check and print function throws an `UnsupportedModelError` and I am not sure why. I was told that they reached out to Star and can test print off of the "Star app". I assume they tested via the StarPRNT SDK app but I am not 100% certain on that.

Additionally, both clients are reporting the printers as being discovered as "unknown". Though from the logs, it seems like one was discovered with the model info available, would a printer being "unknown" cause these issues?

TLDR:
1. Why might their TSP143III (Ethernet) printers be discovered as unknown and can I tell them to adjust some settings to avoid this?
    - I know Bluetooth can have this occur due to a custom name in the settings being set but I have not seen anything similar for the ethernet printers
    - I was also told at least one of the client's factory reset their printer while trying to fix these errors but the printer still shows as unknown
2. Even if discovered as unknown, if I have the identifier and interfaceType, shouldn't print status check and print functionality work as expected and not thrown an `UnsupportedModelError`?
3. As shown from the error logs, is there something I need to do different with the StarPrinter and/or StarConnectionSettings when it is an ethernet printer?

## Your device where the bug occurs 
 - Device: 
    iPad
 - OS:
    17.6.1, 17.7.2

## Your printer
- Model Names:
    TSP143III
- Firmware Version:
    N/A
- Interface:
    Lan (Ethernet)

## Your development environment
ProductName: macOS
ProductVersion: 14.6.1
BuildVersion: 23G93

### Additional Context
These are a couple of errors that we've logged. I am now confused by the report that it is discovered as "unknown" since the model is present but they still receive a bunch of `UnsupportedModelError` when checking the status or trying to print.
- `popStarConnectionSettings` is the settings we store in redux to create a new `StarPrinter`
- `connectionSettings` are the settings of the created `StarPrinter`
- they do have a 2nd printer that is discovered as unknown. I am not 100% if they only expected to discover 1 printer

```
{"name":"StarIO10UnsupportedModelError","_errorCode":0}
[DATA]:
{
  "popStarConnectionSettings": {
    "identifier":"00116207AB56",
    "interfaceType":"Lan",
    "model":"TSP143 (STR_T-001)"
  },
  "connectionSettings":{
    "interfaceType":"Lan",
    "identifier":"00116207AB56",
    "autoSwitchInterface":true
  }
}

{"name":"StarIO10UnsupportedModelError","_errorCode":0}
[DATA]:
{
  "popStarConnectionSettings: { 
    "identifier":"00116208543C",
    "interfaceType":"Lan",
    "model":"Unknown"
  },
  "connectionSettings":{
    "interfaceType":"Lan",
    "identifier":"00116208543C",
    "autoSwitchInterface":true
  }
}
```

**Comment by matthewparavati** [comment]
@can-miki or @bandit-ibayashi, would I be able to get some feedback/insight/response here?

**Comment by can-miki** [comment]
Sorry for the late reply.

We would like to confirm the situation. Could you please provide information on the following points?

Did this issue occur suddenly?
 - Was it working fine before, but the issue started occurring at a certain change or even? Or has the customer been unable to use it from the beginning?

One concern is the possibility that the customer's printer is not "TSP100IIILAN." For example, if it is "TSP100LAN"(a very old model), this library is not supported.
Please refer to [here](https://star-m.jp/products/s_print/sdk/react-native-star-io10/manual/en/index.html) for Supported Models(printer).

We will try printing using a "TSP100IIILAN" on our side. If you find anything or have any updates, please let us know.

**Comment by matthewparavati** [comment]
Hi @can-miki, no worries and thank you for the response. 

According to the clients it stopped working after a recent app update but I am not certain which versions they updated from

We used to use this unofficial package/repo: [Popmenu/react-native-star-prnt](https://github.com/Popmenu/react-native-star-prnt).

We somewhat recently switched over to the official `react-native-star-io10` package, around late summer, early fall.

So I am not certain if it worked with the old package and not the official one or stopped working with the official one after an update 🤦 

Our most recent app updates, should not have included any major or breaking printer changes.

What would I need from the client to confirm if it is a very old "TSP100LAN" model? Would that info be on the "printer config/settings" that you can print out from the printer?

Thank you and will do. We have purchased a TSP100IIILAN printer so we will be able to do some additional testing on our end, too!

**Comment by can-miki** [comment]
Thank you for the sharing detailed information.

The printer model can be checked using either the Self-Printing or the ID label located on the back of the printer. ([Self-Printing instruction page](https://www.star-m.jp/products/s_print/tsp100iiilan/manual/en/basicOperations/testPrint.htm): The steps are the same for both TSP100LAN and TSP100IIILAN.)
We recommend the Self-Printing option as it contains information on printer settings.

Example:
- Self-Printing for TSP100IIILAN
![image](https://github.com/user-attachments/assets/6bbd7a44-0c63-4ab3-b1df-67a1811741aa)
- the ID label & logo for TSP100IIILAN
![image](https://github.com/user-attachments/assets/11fef33a-4071-4363-ad42-364949cca620)

I tested the TSP100IIILAN using the latest official SDK(1.8.0) on our end and confirmed that printing was successful.

Thank you for sharing information about the previous package (unofficial package). I assume that the unofficial package is based on the StarPRNT SDK (our older SDK). Our official package is based on the Xpand SDK, and since it worked with the TSP650II on your side, we believe the migration is not failing.

For now, I would like to start by receiving the printer's Self-Printing and then begin our analysis. Please feel free to reach out if further assistance is needed.

**My testing environment:**
Device: iPad
OS: 15.8.2

Printer Model Names: TSP143IIILAN
Printer Firmware Version: 2.3
Printer Interface: Lan (Ethernet)

react-native info:
> System:
>   OS: macOS 14.6.1
>   CPU: (8) arm64 Apple M2
>   Memory: 420.17 MB / 8.00 GB
>   Shell:
>     version: "5.9"
>     path: /bin/zsh
> Binaries:
>   Node:
>     version: 22.11.0
>     path: /usr/local/bin/node
>   Yarn:
>     version: 1.22.22
>     path: /usr/local/bin/yarn
>   npm:
>     version: 10.9.0
>     path: /usr/local/bin/npm
>   Watchman:
>     version: 2024.07.01.00
>     path: /usr/local/bin/watchman
> Managers:
>   CocoaPods:
>     version: 1.15.2
>     path: /usr/local/bin/pod
> SDKs:
>   iOS SDK:
>     Platforms:
>       - DriverKit 24.1
>       - iOS 18.1
>       - macOS 15.1
>       - tvOS 18.1
>       - visionOS 2.1
>       - watchOS 11.1
>   Android SDK: Not Found
> IDEs:
>   Android Studio: 2024.1 AI-241.18034.62.2412.12266719
>   Xcode:
>     version: 16.1/16B40
>     path: /usr/bin/xcodebuild
> Languages:
>   Java:
>     version: 17.0.11
>     path: /usr/bin/javac
>   Ruby:
>     version: 2.6.10
>     path: /usr/bin/ruby
> npmPackages:
>   "@react-native-community/cli": Not Found
>   react:
>     installed: 18.3.1
>     wanted: 18.3.1
>   react-native:
>     installed: 0.75.4
>     wanted: 0.75.4
>   react-native-macos: Not Found
> npmGlobalPackages:
>   "*react-native*": Not Found
> Android:
>   hermesEnabled: true
>   newArchEnabled: false
> iOS:
>   hermesEnabled: true
>   newArchEnabled: false

**Comment by matthewparavati** [comment]
@can-miki Thanks for the follow up and additional info. Sorry for the delay, I was off for a week.

I was able to test a TSP143IIILAN printer with our app's implementation as well and it worked fine to discover, add, and print as expected, without any issues.

At the moment, I only have the self print info for 1 of the clients but it does say TSP100LAN so that does appear to be the cause for at least one of them but seems quite likely that both are affected by this

![image](https://github.com/user-attachments/assets/626b246c-be70-4838-a5bf-0d1ce63b57c2)

**Comment by can-miki** [comment]
@matthewparavati Thank you for the update!
I was relieved to hear that the checked results were good.
Please feel free to let us know if you have any additional information or concerns.

**Comment by matthewparavati** [comment]
Sorry forgot to come back to this.

We did get confirmation from the our other client that they are also using the older version of the printer so I will go ahead and close this
